### PR TITLE
Add waiting room rules

### PR DESCRIPTION
**Pull request by broswen**
## Description

Add methods for the management of Waiting Room Rules.

## Has your change been tested?

Unit tests have been created to confirm JSON marshaling works and the correct endpoints/methods are called.

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.
  - https://api.cloudflare.com/#waiting-room-list-waiting-room-rules

### CODE REVIEW

**Comment by jacobbednarz** [comment]
thanks for this @broswen 🏆 i've pushed a couple of changes to this that i wanted to highlight as a FYI

- i swapped the methods to the [next gen (or experimental)](https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md) method signatures. this will help us down the track when we revamp the SDK to use the newer formats.
- i swapped `Enabled` to a boolean pointer. based on my cursory investigation, this is using ERE which we've [had issues in Terraform with uninitialised, true and false values for booleans](https://github.com/cloudflare/terraform-provider-cloudflare/pull/1689). the pointer helps us differentiate internally.

feel free to ping me on the internal PR for the docs if you haven't already and i'll merge this once it's in the pipeline.

**Comment by broswen** [comment]
@jacobbednarz docs are live, updated the PR.

File: .changelog/1102.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+waiting room: add support for waiting room rules
```
Comment (jacobbednarz) [inline .changelog/1102.txt]:
```suggestion
waiting_room: add support for waiting room rules
```

**Comment by jacobbednarz** [comment]
thanks @broswen 👏

**Pull request merged**